### PR TITLE
Fixes #19753: Make rule categories editable

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://issues.rudder.io/issues/19753

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewCategoryDetails.elm [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import Html.Events exposing (onClick, onInput)
+import List.Extra
+import List
+import String exposing ( fromFloat)
+import NaturalOrdering exposing (compareOn)
+import ApiCalls exposing (..)
+import ViewTabContent exposing (buildListCategories)
+
+--
+-- This file contains all methods to display the details of the selected rule.
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewCategoryDetails.elm:15]:
```suggestion
-- This file contains all methods to display the details of the selected category.
```

**Review by ncharles — CHANGES_REQUESTED** [review]
Looks great to me. Can you change the comment and self merge?

**Comment by RaphaelGauthier** [comment]
Commit modified

**Comment by RaphaelGauthier** [comment]
PR rebased

**Pull request merged**